state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1215,154 +1209,241 @@ pub mod pallet {
 				request_id,
 			);
 
-			CcmSwapState::GasSwapScheduled { gas_swap_id: swap_id, principal_output_amount }
+			GasSwapState::Scheduled { gas_swap_id }
 		}
 
-		fn process_swap_outcomes(swaps: &[SwapState]) {
-			for swap in swaps {
-				let swap_request_id = swap.swap.swap_request_id;
+		fn refund_failed_swap(swap: Swap) {
+			let swap_request_id = swap.swap_request_id;
 
-				let Some(mut request) = SwapRequests::<T>::take(swap_request_id) else {
-					log_or_panic!("Swap request {swap_request_id} not found");
-					continue;
-				};
+			let Some(request) = SwapRequests::<T>::take(swap_request_id) else {
+				log_or_panic!("Swap request {swap_request_id} not found");
+				return;
+			};
 
-				let Some(output_amount) = swap.final_output else {
-					log_or_panic!("Swap {} is not completed yet!", swap.swap_id());
-					continue;
-				};
+			Self::deposit_event(Event::<T>::SwapRequestCompleted { swap_request_id: request.id });
```

Comment:
Any particular reason for sandwiching this between these two checks instead of emitting it after both?

Reply:
No reason, updating.